### ISSUE TITLE
Failed to retrieve list of speedtest.net servers

### ISSUE THREAD

**Issue by ronilaukkarinen** [sivel/speedtest-cli#155]
I'm using CentOS 6.6.

	[23:20:34] [49.0°C] rolle ~ $ speedtest --version
	0.3.2

	[23:19:38] [48.0°C] rolle ~ $ speedtest
	Retrieving speedtest.net configuration...
	Retrieving speedtest.net server list...
	Failed to retrieve list of speedtest.net servers

**Comment by gregtampa** [comment]
GOt same error on ubuntu
~# ./speedtest-cli --share
Retrieving speedtest.net configuration...
Could not retrieve speedtest.net configuration

**Comment by sivel** [comment]
Please try with the devel branch:

https://raw.githubusercontent.com/sivel/speedtest-cli/devel/speedtest_cli.py

**Comment by KR2615** [comment]
Debian 7.8 also gives an error "Failed to retrieve list of speedtest.net servers"

After downloaded branch listed above:
Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Failed to retrieve list of speedtest.net servers:

 HTTP Error 302: The HTTP server returned a redirect error that would lead to an infinite loop.
The last 30x error message was:
Found
HTTP Error 302: The HTTP server returned a redirect error that would lead to an infinite loop.
The last 30x error message was:
Found

**Comment by tp0** [comment]
Workaround (if you don't want to use install the devel version):

``/usr/local/lib/python2.7/dist-packages/speedtest_cli.py``

Find the user agent under ``# Some global variables we use`` and change it to e.g.:

> user_agent = 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0'

**Comment by KR2615** [comment]
Works fine by me :)

**Comment by alessiodionisi** [comment]
Not working for me with devel script :(

**Comment by FXFman1209** [comment]
I get this error sporadically, definitely not all the time.

**Comment by SudosFTW** [comment]
tp0's workaround does work. I just had to do the modification to the script since it seems speedtest is denying access to the server list now without a valid user agent. you just remove the entire user_agent line and replace it with what's above, and then the script works a treat!

**Comment by intika** [comment]
@tp0 thanks bro !

**Comment by dakleine** [comment]
tp0's workaround worked for me too! Thanks!

**Comment by TomHoss** [comment]
Getting the same error

**Comment by kud** [comment]
Same here.

**Comment by kud** [comment]
https://github.com/sivel/speedtest-cli/issues/155#issuecomment-115133854 workaround works, indeed.

**Comment by buzzijavier** [comment]
:+1:  the workaround works

**Comment by muhasturk** [comment]
same OSX 10.10.4

**Comment by hista** [comment]
> Workaround (if you don't want to install the devel version):

/usr/local/lib/python2.7/dist-packages/speedtest_cli.py

Unfortunatelly I haven't that file in /usr/local/lib/python2.7/dist-packages/:
ls /usr/local/lib/python2.7/dist-packages/:
speedtest_cli-0.3.2-py2.7.egg

So how to workaroud that issue in a such situation please?

**Comment by Zitrax** [comment]
Indeed workaround works as already confirmed. Did anyone get it to work with the devel branch, or is the workaround what need to go into master ?

**Comment by crunchprank** [comment]
I just added ``http://www.speedtest.net/speedtest-servers.php`` to the list of URLs in the script and it fixed it for me (after line 386).

```
def closestServers(client, all=False):
    """Determine the 5 closest speedtest.net servers based on geographic
    distance
    """

    urls = [
	'https://www.speedtest.net/speedtest-servers-static.php',
        'http://c.speedtest.net/speedtest-servers-static.php',
        'http://www.speedtest.net/speedtest-servers.php',
    ]
```

**Comment by hista** [comment]
Could the dev please apply this patch to the package installed with EasyInstall?
Because as explained previously, there is no /usr/local/lib/python2.7/dist-packages/speedtest_cli.py file in this cas to modify to fix it.

**Comment by metafaniel** [comment]
speedtest_cli.py worked for me nicely as downloaded in a CentOS installation, but it didn't work using cygwin on Windows 10. Workaround worked for me too fixing the problem. THANKS.

**Comment by cecchisandrone** [comment]
tp0's workaround worked...thanks!

**Comment by hista** [comment]
But unfortunately can't work when there is no /usr/local/lib/python2.7/dist-packages/speedtest_cli.py file in that folder to modify :-( How to fix it then?

**Comment by cecchisandrone** [comment]
Try with ```locate speedtest_cli.py``` or ```find / -type f -name "speedtest_cli.py"```

**Comment by src386** [comment]
Thanks tp0 :)

**Comment by whizzzkid** [comment]
tp0 u da real MVP

**Comment by sivel** [comment]
The devel version has been pushed to master and released as v0.3.3.  Closing this issue.
